src/test/kotlin/org/jmailen/gradle/kotlinter/functional/PureKotlinFunctionalTest.kt

```diff
@@ -0,0 +1,106 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import com.mkobit.gradle.test.assertj.GradleAssertions.assertThat
+import com.mkobit.gradle.test.kotlin.testkit.runner.build
+import com.mkobit.gradle.test.kotlin.testkit.runner.buildAndFail
+import com.mkobit.gradle.test.kotlin.testkit.runner.info
+import com.mkobit.gradle.test.kotlin.testkit.runner.profile
+import com.mkobit.gradle.test.kotlin.testkit.runner.setupProjectDir
+import com.mkobit.gradle.test.kotlin.testkit.runner.stacktrace
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class PureKotlinFunctionalTest {
+
+    private lateinit var gradleRunner: GradleRunner
+
+    @Before
+    fun setUp() {
+        gradleRunner = GradleRunner.create().apply {
+            withPluginClasspath()
+            withProjectDir(createTempDir(prefix = "kotlinter"))
+            withArguments()
+             info = true
+             stacktrace = true
+             profile = true
+            setupProjectDir {
+                "build.gradle"(content = plugin) {
+                    append(System.lineSeparator())
+                }
+                "settings.gradle"(content = "rootProject.name = 'pure-kotlin-example'${System.lineSeparator()}")
+            }
+        }
+    }
+
+    @After
+    fun tearDown() {
+        gradleRunner.projectDir.deleteRecursively()
+    }
+
+    @Test
+    fun `basic use case with correct formatting`() {
+        gradleRunner.setupProjectDir {
+            "src/main/java/com/example/" {
+                "Main.kt"(content = """
+                    package com.example
+
+                    fun main(args: Array<String>) {
+                        println("Hello, world!")
+                    }
+                """.trimIndent())
```

Comment:
nit: `trimMargin()` often ends up being more reliable with the explicit `|` at the edge.